src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/CoverageVisualisationService.kt

```diff
@@ -71,15 +75,37 @@ class CoverageVisualisationService(private val project: Project) {
             relativeMutations = (coveredMutations.toDouble() / allMutations * 100).roundToInt()
         }
 
-        visualisationService.panelTitleAbsolute.text = "Absolute Test Coverage for ${testReport.UUT}"
-        visualisationService.panelTitleRelative.text = "Relative Test Coverage for ${testReport.UUT}"
+        // Change the values in the table
+        visualisationService.data[4] = testReport.UUT
+        visualisationService.data[5] = "$relativeLines% ($coveredLines/$allLines)"
+        visualisationService.data[6] = "$relativeBranch% ($coveredBranches/$allBranches)"
+        visualisationService.data[7] = "$relativeMutations% ($coveredMutations/ $allMutations)"
+    }
+
+    /**
+     * Creates a new toolWindow tab for the coverage visualisation.
+     */
+    private fun createToolWindowTab() {
+        val visualisationService = project.service<CoverageToolWindowDisplayService>()
+
+        // Remove coverage visualisation from content manager if necessary
+        val toolWindowManager = ToolWindowManager.getInstance(project).getToolWindow("TestGenie")
+        val contentManager = toolWindowManager!!.contentManager
+        for (content in contentManager.contents) {
+            if (content.displayName.equals("Coverage Visualisation")) {
+                contentManager.removeContent(content, true)
+            }
+        }
```

Comment:
Comparing by display name should be avoided - such string comparisons are a code smell. Instead of having this loop, you could keep a reference to the last opened content and simply close that

Reply:
I don't see why it is a problem. "TestGenie" toolwindow was also retrieved by its display name, so it is also compared. Moreover, where are you going to store that last opened content?

Reply:
I am not an SQT TA though

Reply:
This should now be fixed